lib/client/api_test.go

```diff
@@ -499,6 +501,51 @@ func TestApplyProxySettings(t *testing.T) {
 	}
 }
 
+type testAgent struct {
+	agent.Agent
+}
+
+type mockSigner struct {
+	ssh.Signer
```

Comment:
Don't embed the interface here, instead declare the methods you want. We are not setting the embedded signer, nor does it look that is the intent, so this will panic if we call mockSigner.Sign.